core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -1406,10 +1429,13 @@ class Partition(val topicPartition: TopicPartition,
         case Left(error: Errors) => error match {
           case Errors.UNKNOWN_TOPIC_OR_PARTITION =>
             debug(s"Controller failed to update ISR to $proposedIsrState since it doesn't know about this topic or partition. Giving up.")
+            isrChangeMetrics.markFailed()
           case Errors.FENCED_LEADER_EPOCH =>
             debug(s"Controller failed to update ISR to $proposedIsrState since we sent an old leader epoch. Giving up.")
+            isrChangeMetrics.markFailed()
           case Errors.INVALID_UPDATE_VERSION =>
             debug(s"Controller failed to update ISR to $proposedIsrState due to invalid zk version. Giving up.")
+            isrChangeMetrics.markFailed()
           case _ =>
             warn(s"Controller failed to update ISR to $proposedIsrState due to unexpected $error. Retrying.")
             sendAlterIsrRequest(proposedIsrState)
```

Comment:
Note that I'm not including the `markFailed` call here. Think we should include it here? Or possibly introduce a new retry metric?

Reply:
Hmm, I think we should probably do it. If we hit some unexpected case which the broker can't get out of, we'd want the metric to show it.

Reply:
Sounds good